### PR TITLE
coverity fixes

### PR DESCRIPTION
**Pull request by neheb**
I added some builds to https://scan.coverity.com/projects/exiv2 and fixed several issues

### CODE REVIEW

File: src/types.cpp [hunk 1]
```diff
@@ -615,11 +615,11 @@ Rational floatToRationalCast(float f) {
   const double d = f;
   // Beware: primitive conversion algorithm
   int32_t den;
-  if (std::fabs(d) <= std::numeric_limits<int32_t>::max() / 1000000) {
+  if (std::fabs(d) <= std::numeric_limits<int32_t>::max() / 1000000.0) {
```
Comment (kevinbackhouse) [inline src/types.cpp:618]: I think the integer rounding is deliberate here.
```suggestion
  if (std::fabs(d) <= static_cast<double>(std::numeric_limits<int32_t>::max() / 1000000)) {
```

File: src/types.cpp [hunk 2]
```diff
@@ -615,11 +615,11 @@ Rational floatToRationalCast(float f) {
...
     den = 1000000;
-  } else if (std::fabs(d) <= std::numeric_limits<int32_t>::max() / 10000) {
+  } else if (std::fabs(d) <= std::numeric_limits<int32_t>::max() / 10000.0) {
     den = 10000;
-  } else if (std::fabs(d) <= std::numeric_limits<int32_t>::max() / 100) {
+  } else if (std::fabs(d) <= std::numeric_limits<int32_t>::max() / 100.0) {
```
Comment (kevinbackhouse) [inline src/types.cpp:622]: same as above

File: src/version.cpp [hunk 3]
```diff
@@ -166,7 +166,7 @@ static std::vector<std::string> getLoadedLibraries() {
   snprintf(procsz, sizeof(procsz), "/proc/%d/path/a.out", getpid());
   int l = readlink(procsz, pathsz, sizeof(pathsz));
   if (l > 0) {
-    pathsz[l] = '\0';
+    pathsz[l - 1] = '\0';
```
Comment (kevinbackhouse) [inline src/version.cpp:169]: Based on what I've read in the manpage for `readlink`, I think this is the correct way to do it:

```c
ssize_t l = readlink(procsz, pathsz, sizeof(pathsz) - 1);
if (l > 0) {
  pathsz[l] = '\0';
```

Comment (neheb) [reply src/version.cpp]: Fixed.

File: src/types.cpp (hunk 2 above)
Comment (neheb) [reply src/types.cpp]: Removed,

File: src/types.cpp (hunk 1 above)
Comment (neheb) [reply src/types.cpp]: Removed.

**Review by kevinbackhouse — APPROVED** [review]

**Pull request merged**